test/functional/fixtures/screenshots-on-fails/test.js

```diff
@@ -0,0 +1,140 @@
+        return runTests('./testcafe-fixtures/screenshots-on-fails.js', 'Screenshot on error in test code raised',
+            { shouldFail: true, screenshotsOnFails: true, setScreenshotPath: true, only: 'local' })
+            .catch(function (errs) {
+                testErrors = errs;
+                return checkScreenshotsCreated();
+            })
+            .then(function (success) {
+                if (!testOnTravis) {
+                    errorInEachBrowserContains(testErrors, SCREENSHOT_PATH_MESSAGE_TEXT, 0);
+                    expect(success).eql(true);
+                }
+            });
+    });
+
+    it('Should take screenshot on assertion failed', function () {
+        return runTests('./testcafe-fixtures/screenshots-on-fails.js', 'Screenshot on assertion failed',
+            { shouldFail: true, screenshotsOnFails: true, setScreenshotPath: true, only: 'local' })
+            .catch(function (errs) {
+                testErrors = errs;
+                return checkScreenshotsCreated();
+            })
+            .then(function (success) {
+                if (!testOnTravis) {
+                    errorInEachBrowserContains(testErrors, SCREENSHOT_PATH_MESSAGE_TEXT, 0);
+                    expect(success).eql(true);
+                }
+            });
+    });
+
+    it('Should take screenshot on beforeEach raised error', function () {
+        return runTests('./testcafe-fixtures/fail-in-before-each.js', 'Screenshot on beforeEach raised error',
+            { shouldFail: true, screenshotsOnFails: true, setScreenshotPath: true, only: 'local' })
+            .catch(function (errs) {
+                testErrors = errs;
+                return checkScreenshotsCreated();
+            })
+            .then(function (success) {
+                if (!testOnTravis) {
+                    errorInEachBrowserContains(testErrors, SCREENSHOT_PATH_MESSAGE_TEXT, 0);
+                    expect(success).eql(true);
+                }
+            });
+    });
+
+    it('Should take screenshots on test error and on afterEach raised error', function () {
+        return runTests('./testcafe-fixtures/fail-in-test-and-after-each.js', 'Screenshot on test error and on afterEach raised error',
+            { shouldFail: true, screenshotsOnFails: true, setScreenshotPath: true, only: 'local' })
+            .catch(function (errs) {
+                testErrors = errs;
+                return checkScreenshotsCreated(false, 4);
+            })
+            .then(function (success) {
+                if (!testOnTravis) {
+                    errorInEachBrowserContains(testErrors, SCREENSHOT_PATH_MESSAGE_TEXT, 0);
+                    errorInEachBrowserContains(testErrors, SCREENSHOT_PATH_MESSAGE_TEXT, 1);
+                    expect(success).eql(true);
+                }
+            });
+    });
+
+    it('Should not take screenshot on ensure action element fails without screenshotsOnFails flag', function () {
```

Comment:
it('Should not take a screenshot if the ensureElement method fails with no screenshotsOnFails flag set', function () {